bin/run.sh

```diff
@@ -20,41 +20,57 @@ if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
     echo "usage: ./bin/run.sh exercise-slug path/to/solution/folder/ path/to/output/directory/"
     exit 1
 fi
-
 slug="$1"
+work_dir=$(pwd)
 solution_dir=$(realpath "${2%/}")
 output_dir=$(realpath "${3%/}")
 results_file="${output_dir}/results.json"
 
 # Create the output directory if it doesn't exist
 mkdir -p "${output_dir}"
 
-echo "${slug}: testing..."
 
-# Run the tests for the provided implementation file and redirect stdout and
-# stderr to capture it
-test_output=$(false)
-# TODO: substitute "false" with the actual command to run the test:
-# test_output=$(command_to_run_tests 2>&1)
+# Run the tests for the provided implementation file
+cd "$solution_dir" || exit
+# Output is saved at this location
+echo "==== ${slug}: testing..."
+test_output=""
+if ! [ -s *.bal ]; then
+    test_output="WARNING: student did not add code to the balllerina file."
+fi
+if [ ! -e Ballerina.toml ]; then
+    test_output="$test_output \n WARNING: student did not upload Ballerina.toml."
+fi
+if [ ! -e Dependencies.toml ]; then
+    test_output="$test_output \n WARNING: student did not upload Dependencies.toml."
+fi
```

Comment:
Note that these conditions should never match.
What the test runner receives is not just what the student submitted, but a "merge" of the exercise dir's contents as defined in the track repo with and student-submitted files added/overwriting.

What I would thus suggest is to add the `Ballerina.toml` and `Dependencies.toml` files to all the tests in the `tests/*` directory. That makes it a more representative situation.

Reply:
Ok, that makes it easier. I will update to assume these exist. The build command will report issues with these files.